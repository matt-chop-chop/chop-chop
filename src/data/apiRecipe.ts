export const apiRecipe: ApiRecipe = {
  idMeal: "52922",
  strMeal: "Prawn & Fennel Bisque",
  strDrinkAlternate: null,
  strCategory: "Side",
  strArea: "French",
  strInstructions:
    "Shell the prawns, then fry the shells in the oil in a large pan for about 5 mins. Add the onion, fennel and carrots and cook for about 10 mins until the veg start to soften. Pour in the wine and brandy, bubble hard for about 1 min to drive off the alcohol, then add the tomatoes, stock and paprika. Cover and simmer for 30 mins. Meanwhile, chop the prawns.\r\nBlitz the soup as finely as you can with a stick blender or food processor, then press through a sieve into a bowl. Spend a bit of time really working the mixture through the sieve as this will give the soup its velvety texture.\r\nTip back into a clean pan, add the prawns and cook for 10 mins, then blitz again until smooth. You can make and chill this a day ahead or freeze it for 1 month. Thaw ovenight in the fridge. To serve, gently reheat in a pan with the cream. If garnishing, cook the 8 prawns in a little butter. Spoon into small bowls and top with the prawns and snipped fennel fronds.",
  strMealThumb:
    "https://www.themealdb.com/images/media/meals/rtwwvv1511799504.jpg",
  strTags: "Soup,Warm,Seafood,Shellfish",
  strYoutube: "https://www.youtube.com/watch?v=4thpMbDakgM",
  strIngredient1: "Tiger Prawns",
  strIngredient2: "Olive Oil",
  strIngredient3: "Onion",
  strIngredient4: "Fennel",
  strIngredient5: "Carrots",
  strIngredient6: "Dry White Wine",
  strIngredient7: "Brandy",
  strIngredient8: "Chopped Tomatoes",
  strIngredient9: "Fish Stock",
  strIngredient10: "Paprika",
  strIngredient11: "Double Cream",
  strIngredient12: "Prawns",
  strIngredient13: "",
  strIngredient14: "",
  strIngredient15: "",
  strIngredient16: "",
  strIngredient17: "",
  strIngredient18: "",
  strIngredient19: "",
  strIngredient20: "",
  strMeasure1: "450g",
  strMeasure2: "4 tbs",
  strMeasure3: "1 large",
  strMeasure4: "1 large",
  strMeasure5: "2 chopped",
  strMeasure6: "150ml",
  strMeasure7: "1 tbs",
  strMeasure8: "400g",
  strMeasure9: "1L",
  strMeasure10: "2 pinches",
  strMeasure11: "150ml",
  strMeasure12: "8",
  strMeasure13: "",
  strMeasure14: "",
  strMeasure15: "",
  strMeasure16: "",
  strMeasure17: "",
  strMeasure18: "",
  strMeasure19: "",
  strMeasure20: "",
  strSource:
    "https://www.bbcgoodfood.com/recipes/71600/prawn-and-fennel-bisque",
  strImageSource: null,
  strCreativeCommonsConfirmed: null,
  dateModified: null,
};
