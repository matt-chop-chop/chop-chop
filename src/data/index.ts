export { recipe } from "./recipe";
export { apiRecipe } from "./apiRecipe";
