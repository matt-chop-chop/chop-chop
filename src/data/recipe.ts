export const recipe: Recipe = {
  id: "52922",
  area: "French",
  category: "Side",
  image: "https://www.themealdb.com/images/media/meals/rtwwvv1511799504.jpg",
  ingredients: [
    {
      name: "Tiger Prawns",
      measurement: "450g",
      image: "https://www.themealdb.com/images/ingredients/Tiger Prawns.png",
    },
    {
      name: "Olive Oil",
      measurement: "4 tbs",
      image: "https://www.themealdb.com/images/ingredients/Olive Oil.png",
    },
    {
      name: "Onion",
      measurement: "1 large",
      image: "https://www.themealdb.com/images/ingredients/Onion.png",
    },
    {
      name: "Fennel",
      measurement: "1 large",
      image: "https://www.themealdb.com/images/ingredients/Fennel.png",
    },
    {
      name: "Carrots",
      measurement: "2 chopped",
      image: "https://www.themealdb.com/images/ingredients/Carrots.png",
    },
    {
      name: "Dry White Wine",
      measurement: "150ml",
      image: "https://www.themealdb.com/images/ingredients/Dry White Wine.png",
    },
    {
      name: "Brandy",
      measurement: "1 tbs",
      image: "https://www.themealdb.com/images/ingredients/Brandy.png",
    },
    {
      name: "Chopped Tomatoes",
      measurement: "400g",
      image:
        "https://www.themealdb.com/images/ingredients/Chopped Tomatoes.png",
    },
    {
      name: "Fish Stock",
      measurement: "1L",
      image: "https://www.themealdb.com/images/ingredients/Fish Stock.png",
    },
    {
      name: "Paprika",
      measurement: "2 pinches",
      image: "https://www.themealdb.com/images/ingredients/Paprika.png",
    },
    {
      name: "Double Cream",
      measurement: "150ml",
      image: "https://www.themealdb.com/images/ingredients/Double Cream.png",
    },
    {
      name: "Prawns",
      measurement: "8",
      image: "https://www.themealdb.com/images/ingredients/Prawns.png",
    },
  ],
  instructions: [
    "Shell the prawns, then fry the shells in the oil in a large pan for about 5 mins. ",
    "Add the onion, fennel and carrots and cook for about 10 mins until the veg start to soften. ",
    "Pour in the wine and brandy, bubble hard for about 1 min to drive off the alcohol, then add the tomatoes, stock and paprika. ",
    "Cover and simmer for 30 mins. ",
    "Meanwhile, chop the prawns.\r\nBlitz the soup as finely as you can with a stick blender or food processor, then press through a sieve into a bowl. ",
    "Spend a bit of time really working the mixture through the sieve as this will give the soup its velvety texture.\r\nTip back into a clean pan, add the prawns and cook for 10 mins, then blitz again until smooth. ",
    "You can make and chill this a day ahead or freeze it for 1 month. ",
    "Thaw ovenight in the fridge. ",
    "To serve, gently reheat in a pan with the cream. ",
    "If garnishing, cook the 8 prawns in a little butter. ",
    "Spoon into small bowls and top with the prawns and snipped fennel fronds.",
  ],
  name: "Prawn & Fennel Bisque",
  thumbnail:
    "https://www.themealdb.com/images/media/meals/rtwwvv1511799504.jpg/preview",
  source: "https://www.bbcgoodfood.com/recipes/71600/prawn-and-fennel-bisque",
  tags: ["Soup", "Warm", "Seafood", "Shellfish"],
  ingredientCount: 12,
  time: 75,
  youtube: "https://www.youtube.com/watch?v=4thpMbDakgM",
};
