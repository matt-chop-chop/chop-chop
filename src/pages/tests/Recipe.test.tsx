import { shallow } from "enzyme";
import React from "react";
import { apiRecipe as mockApiRecipe } from "@/data";
import Recipe from "../recipe/[id].page";

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn().mockReturnValue({ query: { id: "123" } }),
}));

jest.mock("react-query", () => ({
  useQuery: jest.fn().mockReturnValue({
    data: {
      data: { meals: [mockApiRecipe] },
    },
    isLoading: false,
    isSuccess: true,
  }),
}));

describe("Recipe", () => {
  it("is exported from Recipe", () => {
    expect(Recipe).toBeDefined();
  });

  it("renders as expected", () => {
    const wrapper = shallow(<Recipe />);

    expect(wrapper).toMatchSnapshot();
  });
});
