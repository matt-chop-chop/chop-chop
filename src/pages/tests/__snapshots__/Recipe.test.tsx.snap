// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe renders as expected 1`] = `
<Flex
  flexDirection="column"
>
  <RecipeHeader
    recipe={
      {
        "area": "French",
        "category": "Side",
        "id": "52922",
        "image": "https://www.themealdb.com/images/media/meals/rtwwvv1511799504.jpg",
        "ingredientCount": 12,
        "ingredients": [
          {
            "image": "https://www.themealdb.com/images/ingredients/Tiger Prawns.png",
            "measurement": "450g",
            "name": "Tiger Prawns",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Olive Oil.png",
            "measurement": "4 tbs",
            "name": "Olive Oil",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Onion.png",
            "measurement": "1 large",
            "name": "Onion",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Fennel.png",
            "measurement": "1 large",
            "name": "Fennel",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Carrots.png",
            "measurement": "2 chopped",
            "name": "Carrots",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Dry White Wine.png",
            "measurement": "150ml",
            "name": "Dry White Wine",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Brandy.png",
            "measurement": "1 tbs",
            "name": "Brandy",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Chopped Tomatoes.png",
            "measurement": "400g",
            "name": "Chopped Tomatoes",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Fish Stock.png",
            "measurement": "1L",
            "name": "Fish Stock",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Paprika.png",
            "measurement": "2 pinches",
            "name": "Paprika",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Double Cream.png",
            "measurement": "150ml",
            "name": "Double Cream",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Prawns.png",
            "measurement": "8",
            "name": "Prawns",
          },
        ],
        "instructions": [
          "Shell the prawns, then fry the shells in the oil in a large pan for about 5 mins. ",
          "Add the onion, fennel and carrots and cook for about 10 mins until the veg start to soften. ",
          "Pour in the wine and brandy, bubble hard for about 1 min to drive off the alcohol, then add the tomatoes, stock and paprika. ",
          "Cover and simmer for 30 mins. ",
          "Meanwhile, chop the prawns.
Blitz the soup as finely as you can with a stick blender or food processor, then press through a sieve into a bowl. ",
          "Spend a bit of time really working the mixture through the sieve as this will give the soup its velvety texture.
Tip back into a clean pan, add the prawns and cook for 10 mins, then blitz again until smooth. ",
          "You can make and chill this a day ahead or freeze it for 1 month. ",
          "Thaw ovenight in the fridge. ",
          "To serve, gently reheat in a pan with the cream. ",
          "If garnishing, cook the 8 prawns in a little butter. ",
          "Spoon into small bowls and top with the prawns and snipped fennel fronds.",
        ],
        "name": "Prawn & Fennel Bisque",
        "source": "https://www.bbcgoodfood.com/recipes/71600/prawn-and-fennel-bisque",
        "tags": [
          "Soup",
          "Warm",
          "Seafood",
          "Shellfish",
        ],
        "thumbnail": "https://www.themealdb.com/images/media/meals/rtwwvv1511799504.jpg/preview",
        "time": 75,
        "youtube": "https://www.youtube.com/watch?v=4thpMbDakgM",
      }
    }
  />
  <Flex
    flexDirection="column"
    p={4}
  >
    <IngredientList
      ingredients={
        [
          {
            "image": "https://www.themealdb.com/images/ingredients/Tiger Prawns.png",
            "measurement": "450g",
            "name": "Tiger Prawns",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Olive Oil.png",
            "measurement": "4 tbs",
            "name": "Olive Oil",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Onion.png",
            "measurement": "1 large",
            "name": "Onion",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Fennel.png",
            "measurement": "1 large",
            "name": "Fennel",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Carrots.png",
            "measurement": "2 chopped",
            "name": "Carrots",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Dry White Wine.png",
            "measurement": "150ml",
            "name": "Dry White Wine",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Brandy.png",
            "measurement": "1 tbs",
            "name": "Brandy",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Chopped Tomatoes.png",
            "measurement": "400g",
            "name": "Chopped Tomatoes",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Fish Stock.png",
            "measurement": "1L",
            "name": "Fish Stock",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Paprika.png",
            "measurement": "2 pinches",
            "name": "Paprika",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Double Cream.png",
            "measurement": "150ml",
            "name": "Double Cream",
          },
          {
            "image": "https://www.themealdb.com/images/ingredients/Prawns.png",
            "measurement": "8",
            "name": "Prawns",
          },
        ]
      }
    />
    <Method
      instructions={
        [
          "Shell the prawns, then fry the shells in the oil in a large pan for about 5 mins. ",
          "Add the onion, fennel and carrots and cook for about 10 mins until the veg start to soften. ",
          "Pour in the wine and brandy, bubble hard for about 1 min to drive off the alcohol, then add the tomatoes, stock and paprika. ",
          "Cover and simmer for 30 mins. ",
          "Meanwhile, chop the prawns.
Blitz the soup as finely as you can with a stick blender or food processor, then press through a sieve into a bowl. ",
          "Spend a bit of time really working the mixture through the sieve as this will give the soup its velvety texture.
Tip back into a clean pan, add the prawns and cook for 10 mins, then blitz again until smooth. ",
          "You can make and chill this a day ahead or freeze it for 1 month. ",
          "Thaw ovenight in the fridge. ",
          "To serve, gently reheat in a pan with the cream. ",
          "If garnishing, cook the 8 prawns in a little butter. ",
          "Spoon into small bowls and top with the prawns and snipped fennel fronds.",
        ]
      }
    />
    <Box
      border="1px solid var(--chakra-colors-light-text)"
      mt="80px"
    >
      <AspectRatio
        maxWidth="100%"
        ratio={1.7777777777777777}
      >
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          allowFullScreen={true}
          src="https://www.youtube.com/embed/4thpMbDakgM"
          title="Prawn & Fennel Bisque tutorial video"
        />
      </AspectRatio>
    </Box>
    <Link
      _hover={
        {
          "transform": "scale(1.1)",
          "transition": "transform 250ms ease-in",
        }
      }
      as={
        {
          "$$typeof": Symbol(react.forward_ref),
          "default": [Circular],
          "render": [Function],
        }
      }
      href="https://www.bbcgoodfood.com/recipes/71600/prawn-and-fennel-bisque"
      isExternal={true}
      mt={8}
      transition="transform 250ms ease-out"
      width="fit-content"
    >
      <Text
        variant="body"
      >
         Click to see the original recipe source
      </Text>
    </Link>
  </Flex>
</Flex>
`;
